benchmark/src/main.rs

```diff
@@ -0,0 +1,335 @@
+mod cli;
+mod error;
+mod harness;
+mod io;
+mod ir_utils;
+mod logger;
+mod runner;
+mod stats;
+use crate::cli::*;
+use crate::error::HarvestResult;
+use crate::harness::*;
+use crate::io::*;
+use crate::ir_utils::{cargo_build_result, raw_cargo_package, raw_source};
+use crate::logger::TeeLogger;
+use crate::stats::*;
+use clap::Parser;
+use harvest_ir::HarvestIR;
+use harvest_translate::cli::initialize;
+use harvest_translate::transpile;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+/// Encapsulate important results from transpilation
+pub struct TranspilationResult {
+    translation_success: bool,
+    build_success: bool,
+    rust_binary_path: PathBuf,
+    build_error: Option<String>,
+}
+
+impl TranspilationResult {
+    /// Extract relevant info from HarvestIR
+    pub fn from_ir(ir: &HarvestIR) -> Self {
+        let translation_success = raw_cargo_package(ir).is_ok();
+        let (build_success, rust_binary_path, build_error) = match cargo_build_result(ir) {
+            Ok(artifacts) => (true, artifacts[0].clone(), None), // should check that there is only 1 artifact
+            Err(err) => (false, PathBuf::new(), Some(err)),
+        };
+
+        Self {
+            translation_success,
+            build_success,
+            rust_binary_path,
+            build_error,
+        }
+    }
+}
+
+/// Translates a C source directory to a Rust Cargo project using harvest_translate
+pub async fn translate_c_directory_to_rust_project(
```

Comment:
Why is this `async`?

Reply:
Actually... I removed all `async`/`await`/`tokio` references from this crate and it seems to work fine. Is this a leftover from an earlier iteration that was concurrent?